### DISCUSSION TITLE
directive support for document 360-style callouts

### DISCUSSION THREAD

**Discussion by parkerjgit** [remarkjs/remark#1078] [Q&A] [answered]
Hi, 

[document 360 callouts](https://docs.document360.com/docs/the-markdown-editor) look like this:

:::(Info) (Title)
Here is some info...
:::

:::(Warning) (Title)
Here is a warning...
:::

Does remark-directive or other plugin support this syntax? or any recommendations?

**Comment by ChristianMurphy** [comment, marked as answer]
[`remark-directive`](https://github.com/remarkjs/remark-directive) supports the [generic directive proposed standard](https://talk.commonmark.org/t/generic-directives-plugins-syntax/444).
document 360 has different syntax, and is not supported as a directive.

You are welcome to look for an existing plugin which parses this syntax.
* https://github.com/remarkjs/remark/blob/main/doc/plugins.md#list-of-plugins
* https://github.com/topics/remark-plugin

I am not aware of one.

> or any recommendations?

The general recommendations are here: https://github.com/micromark/micromark#extending-markdown
The shortest and best advise is nudge products towards using standard commonmark or gfm, instead of inventing yet another flavor of markdown.

If the syntax is absolutely needed, https://github.com/micromark/micromark#creating-a-micromark-extension gives an overview of extending syntax, it can be a rather involved process, as markdown and how all of its different features interact can be quite complex.
